### PR TITLE
feat: Add table entries to properties panel

### LINKED ISSUE #890: Create properties panels for the table component
### TODO

- [ ] Add section `General`, with the following fields:
   - [ ] `Table header`: optional, type text, FEEL toggable
   - [ ] `Data source`: required, type text, FEEL toggable
   - [ ] `Default number of rows`: required, type number, greater than 0, default value 10
   - [ ] `Pagination`: required, toggle, enabled by default
- [ ] Add section `Columns`, FEEL toggable, list of objects `header` and `context key`

### Additional information
- [Figma link](https://www.figma.com/file/J78Djf3PgCEAPxFD86floX/Form-Editor-Components?type=design&node-id=290-68794&mode=design&t=0dXyhVaRNNXSJBTH-0)

### PR DESCRIPTION
**Pull request by vsgoulart**
Closes #890

- [ ] This PR adds a new `form-js` element or visually changes an existing component.
  * => In that case, we need to ensure we follow up on this, e.g. by [creating an issue in Tasklist](https://github.com/camunda/tasklist/issues/new/choose)

### CODE REVIEW

File: packages/form-js-editor/src/features/properties-panel/entries/TextEntry.js [hunk 1]
```diff
@@ -25,17 +24,6 @@ export default function TextEntry(props) {
     }
   ];
 
-  // todo: skipped to make the release without too much risk
-  // if (templating.isTemplate(field.text)) {
-  //   entries.push(simpleBoolEntryFactory({
-  //     id: 'strict',
```
Comment (vsgoulart) [inline packages/form-js-editor/src/features/properties-panel/entries/TextEntry.js:31]: @Skaiir I deleted these comments, should I create an issue to replace them?

File: packages/form-js-editor/src/features/properties-panel/entries/ColumnEntry.js
Comment (vsgoulart) [inline packages/form-js-editor/src/features/properties-panel/entries/ColumnEntry.js:1]: The logic of the the static column headers is sort of similar with the static options in the select, but I think they're not that related to be worth it to create a common reusable component

File: packages/form-js-editor/src/features/properties-panel/entries/TableDataSourceEntry.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+export function TableDataSourceEntry(props) {
+  const {
+    editField,
+    field
+  } = props;
+
+  const entries = [];
+  entries.push({
+    id: 'source',
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/entries/TableDataSourceEntry.js:17]: Maybe `dataSource`?

File: packages/form-js-editor/src/features/properties-panel/entries/LabelEntry.js [hunk 3]
```diff
@@ -51,13 +51,29 @@ export default function LabelEntry(props) {
...
 
   return entries;
 }
 
+
+/**
+ * @param {string} type
+ * @returns {string}
+ */
+function getEntryLabel(type) {
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/entries/LabelEntry.js:66]: I added exactly that for the `iFrame`, so you probably need to rebase the branch.

File: packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  };
+
+  /**
+   * @param {string|void} value
+   */
+  const setValue = (value) => {
+    switch (value) {
+    case OPTIONS.static.value:
+      editField(field, {
+        columnsExpression: undefined,
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js:74]: For `values` we have a [modeling behavior](https://github.com/bpmn-io/form-js/blob/develop/packages/form-js-editor/src/features/modeling/behavior/ValuesSourceBehavior.js#L29) that removes the other properties. The advantage of that approach is that it also does the clean ups when using the modeling API directly, and not only in the properties panel entry. It's also pluggable as an extension can simply remove that behavior (for whatever reason). 

Wdyt, should we do the same for `columnsExpression` and `columns` or only once it becomes more complex with more possibilities do define the source?

File: packages/form-js-editor/src/features/properties-panel/groups/TableHeaderGroups.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      ...StaticColumnsSourceEntry({ field, editField, id })
+    });
+  }
+
+  return groups;
+}
+
+// helpers //////////
+
+const TOOLTIP_TEXT = `"Static" defines a constant, predefined set of form options.
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/groups/TableHeaderGroups.js:55]:
```suggestion
const TOOLTIP_TEXT = `"List of items" defines a constant, predefined set of form options.
```

File: packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js (hunk 4 above)
Comment (vsgoulart) [reply packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js:74]: Yeah, I was waiting for you to raise this topic. I'll take a look into this tomorrow

File: packages/form-js-editor/src/features/properties-panel/entries/TextEntry.js (hunk 1 above)
Comment (Skaiir) [reply packages/form-js-editor/src/features/properties-panel/entries/TextEntry.js:31]: Nah we've been good without it, if it's needed it'll come up again.

**Comment by Skaiir** [comment]
Had a look over, agreeing with Niklas's points but that's about all I saw at a glance. Will have a serious long review on the complete PR.

File: packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js (hunk 4 above)
Comment (vsgoulart) [reply packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js]: @pinussilvestrus Should be done now

Comment (pinussilvestrus) [reply packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js]: Thanks for the follow up, looks good to me 👍

File: packages/form-js-editor/test/spec/features/modeling/behavior/ColumnsSourceBehavior.spec.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+describe('features/modeling - ColumnsSourceBehavior', function() {
+
+  beforeEach(bootstrapFormEditor(schema, {
+    additionalModules: [
+      modelingModule
+    ]
+  }));
+
+
+  it('should NOT remove values source properties', inject(function(formFieldRegistry, modeling) {
```
Comment (pinussilvestrus) [inline packages/form-js-editor/test/spec/features/modeling/behavior/ColumnsSourceBehavior.spec.js:19]:
```suggestion
  it('should NOT remove columns source properties', inject(function(formFieldRegistry, modeling) {
```

**Review by pinussilvestrus — APPROVED** [review]

**Pull request merged**